admiral/envs/components/wrappers/observer_wrapper.py

```diff
@@ -0,0 +1,112 @@
+
+from gym.spaces import Dict, Discrete
+import numpy as np
+
+from admiral.envs.components.agent import PositionAgent, AgentObservingAgent, ObservingAgent
+
+def obs_filter_step(distance, view):
```

Comment:
Generalize this to receive the two agent objects + any kwargs and then use their parameters from there to determine the probability.